qemu/tests/migration_multi_host_timedrift.py

```diff
@@ -55,12 +56,32 @@ def __init__(self, test, params, env):
             self.sync = SyncData(self.master_id(), self.hostid, self.hosts,
                                  self.id, self.sync_server)
 
+        @error.context_aware
         def check_diff(self, mig_data):
             logging.debug("Sleep 10s")
             time.sleep(10)
             time_drifted = False
             for vm in mig_data.vms:
                 session = vm.wait_for_login()
+
+                if self.is_src and params.get("os_type") == "linux":
+                    error.context("Check the clocksource in guest.",
+                                  logging.info)
+                    check_clocksource_cmd = params.get("check_clocksource_cmd")
+                    clocksource = params.get("clocksource", "kvm-clock")
+                    current_clocksource = session.cmd(check_clocksource_cmd)
+                    logging.debug("current_clocksource in guest is: '%s'" %
+                                  current_clocksource)
+                    if clocksource == "kvm-clock":
+                        s = "kvm-clock" in current_clocksource
+                    else:
+                        s = "kvm-clock" not in current_clocksource
```

Comment:
The logic is strange,
why not compare clocksource with current_clocksource directly.